modyn/config/schema/pipeline-schema.yaml

```diff
@@ -309,6 +309,76 @@ properties:
           Configuration dictionary that will be passed to the trigger on initialization.
     required:
       - id
+  evaluation:
+    type: object
+    description: |
+      Evaluation configuration.
+    properties:
+      datasets:
+        type: array
+        description: |
+          An array of all datasets on which the model is evaluated.
+        minItems: 1
+        items:
+          type: object
+          description: |
+            Configuration for the evaluation dataset.
+          properties:
+            dataset_id:
+              type: string
+              description: |
+                The id of the dataset.
+            batch_size:
+              type: number
+              description: |
+                The batch size to be used during evaluation.
+            dataloader_workers:
+              type: number
+              description: |
+                The number of data loader workers on the evaluation node that fetch data from storage.
+            metrics:
+              type: array
+              description: |
+                All metrics used to evaluate the model on the given dataset.
+              minItems: 1
+              items:
+                type: object
+                description: |
+                  Confguration for the metric.
+                properties:
+                  name:
+                    type: string
+                    description: |
+                      The name of the evaluation metric. Currently we support F1-score, Accuracy and ROC-AUC.
+                  config:
+                    type: object
+                    description: |
+                      Optional configuration for the evaluation metric.
+                  evaluation_transformer_function:
+                    type: string
+                    description: |
+                      The (optional) function used to transform the labels and the model output before evaluation.
```

Comment:
Was there a reason we pass the label through this? Or should we just always pass the label hard coded to the metric? I remember that we need to pass the model output through this, but do not remember the reason for the label

Reply:
Yes, you are right. I initially thought that we will also need to transform the label for some metrics. However, this should not be necessary.